### PR TITLE
[FIX] Localized payment dates in invoice reports

### PR DESCRIPTION
**Pull request by remi-filament**
Odoo PR #34324 
Fix the Odoo issue #34317 by making the payment date to be localized thanks to date widget

Description of the issue/feature this PR addresses:
Fix the issue #34317 by making the payment date to be localized thanks to date widget

Current behavior before PR:
Payment dates not localized in invoice reports

Desired behavior after PR is merged:
Payment dates localized as per Odoo settings in invoice reports

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/views/report_invoice.xml [hunk 1]
```diff
@@ -200,7 +200,7 @@
                 <t t-foreach="payments_vals" t-as="payment_vals">
                     <tr>
                         <td>
-                            <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']"/></i>
+                            <i class="oe_form_field text-right oe_payment_label">Paid on <t t-esc="payment_vals['date']" t-options="{&quot;widget&quot;: &quot;date&quot;}"/></i>
```
Comment (pedrobaeza) [inline addons/account/views/report_invoice.xml]: Why not putting simple quotes instead of `&quot;` for better reading?

**Comment by gdgellatly** [comment]
@pedrobaeza because it is json not python.

**Comment by pedrobaeza** [comment]
@gdgellatly I don't think that's a problem.

**Comment by gdgellatly** [comment]
@pedrobaeza it most definitely can be, although I tend to use python dict() constructor syntax now for these json dicts.

**Comment by pedrobaeza** [comment]
I'm only talking about the readability of the XML file.

It's better to see
```
... t-options="{'widget': 'date'}"/></i>
```
than
```
... t-options="{&quot;widget&quot;: &quot;date&quot;}"/></i>
```
and both should work.

**Comment by gdgellatly** [comment]
> I'm only talking about the readability of the XML file.
> 
> It's better to see
> 
> ```
> ... t-options="{'widget': 'date'}"/></i>
> ```
> 
> than
> 
> ```
> ... t-options="{&quot;widget&quot;: &quot;date&quot;}"/></i>
> ```
> 
> and both should work.
@pedrobaeza 

Both should work, (in v12) because when they removed t-field-options and t-esc-options it also started accepting standard python dicts. But if it needs backporting any further it would need to change. Anyway, as I say, these days I tend toward

`t-options="dict(widget='date')"`

as I find it more readable. But also

`t-options='{"widget": "date"}'` weirdly works and that wasn't uncommon in v11 to avoid the escaping.

File: addons/account/views/report_invoice.xml (hunk 1 above)
Comment (remi-filament) [reply addons/account/views/report_invoice.xml]: I changed it to t-options='{"widget": "date"}' as suggested to keep it consistent with other t-options.

**Review by pedrobaeza — APPROVED** [review]

**Comment by remi-filament** [comment]
Fixed in Odoo https://github.com/odoo/odoo/commit/fdbabfdc59d2d8deb62d08c7bd132a93a5f2f738

**Pull request closed without merging**